api/server/backend/build/backend.go

```diff
@@ -0,0 +1,75 @@
+package build
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/docker/distribution/reference"
+	"github.com/docker/docker/api/types"
+	backendtypes "github.com/docker/docker/api/types/backend"
+	"github.com/docker/docker/builder"
+	"github.com/docker/docker/builder/dockerfile"
+	"github.com/docker/docker/image"
+	"github.com/docker/docker/pkg/stringid"
+	"github.com/pkg/errors"
+	"golang.org/x/net/context"
+)
+
+// ImageComponent provides an interface for working with images
+type ImageComponent interface {
+	SquashImage(from string, to string) (string, error)
+	TagImageWithReference(image.ID, reference.Named) error
+}
+
+// Backend provides build functionality to the API router
+type Backend struct {
+	manager        *dockerfile.BuildManager
+	imageComponent ImageComponent
+}
+
+// NewBackend creates a new build backend from components
+func NewBackend(components ImageComponent, backend builder.Backend) *Backend {
```

Comment:
Just rename. `backend` is meaningless about the behavior so I skipped that, `builderBackend` is fine if you prefer that. I'm not sure why `GetImage()` shouldn't be in the dependency interface for the builder and should be split in the future. It is how `FROM/--from` and default env is implemented.